Lib/test/test_capi/test_misc.py

```diff
@@ -401,6 +401,43 @@ def check_negative_refcount(self, code):
                          br'_Py_NegativeRefcount: Assertion failed: '
                          br'object has negative ref count')
 
+    def run_iter_api_test(self, next_func):
+        dataset = (
+            (), (1,2,3),
+            [], [1,2,3],
+        )
+
+        for input_ in dataset:
```

Comment:
You might use with `self.subTest(input=input_)` to get more info if a test fails.

Reply:
Yes, subTest is nice.